tee-worker/omni-executor/ethereum-rpc/src/lib.rs

```diff
@@ -100,6 +102,24 @@ impl RpcProvider for AlloyRpcProvider {
 			.map_err(|e| error!("Could not get balance: {:?}", e))
 	}
 
+	async fn get_pending_nonce(&self, address: Self::Addr) -> Result<u64, ()> {
+		let provider = ProviderBuilder::new().connect_http(
+			self.url.parse().map_err(|e| error!("Could not parse rpc url: {:?}", e))?,
+		);
+		let transaction_count = self
+			.get_transaction_count(address)
+			.await
+			.map_err(|e| error!("Could not get transaction count: {:?}", e))?;
+
+		let pending_transaction_count = provider
+			.get_transaction_count(address)
+			.block_id(BlockId::Number(BlockNumberOrTag::Pending))
```

Comment:
Are you sure this returns pending transactions count ?
Not overall transaction count (but based on pedning - not included - block)

Reply:
Yeah, you're right, my bad 😅 , I've updated it to only use pending transaction count.